### DISCUSSION TITLE
How to switch to the context of specific webview?

### DISCUSSION THREAD

**Discussion by daniel-a126** [redhat-developer/vscode-extension-tester#1493] [Q&A]
Editor and sidebars can be contributing webviews to VS Code. Editor seems to be contributing only one _active_ at a time, but for sidebars this is a completely different story - there can be multiple and each can have many panels. Sidebar webviews are not discarded when sidebar is hidden. There are therefore multiple webviews at the same time.

As far as I know ([via wiki](https://github.com/redhat-developer/vscode-extension-tester/wiki/WebView)) we have the following API awailable:

```
const webview = new WebView();
await webview.switchToFrame();
...
await webview.switchBack(); 
```

This seems to return the first found webview (typically from Editor).

Is there a way to tell the tester which webview am I interested in? For example by passing a **ViewSection**?

Thank you!

**Comment by djelinek** [comment]
unfortunately there is no identifier for webviews to be more specific which to use. There is open issue for improvement of this -https://github.com/redhat-developer/vscode-extension-tester/issues/1486

The idea is to allow users at least to specify webview title/name which they would like to try to switch to. Is this something what would also help you?

**Comment by daniel-a126** [comment]
Yeah, that would definitely help. Thank you!
